CMakeLists.txt

```diff
@@ -405,13 +498,13 @@ target_compile_definitions(onnx_proto PRIVATE ${ONNX_API_DEFINE})
 
 if(ONNX_USE_LITE_PROTO)
   if(TARGET protobuf::libprotobuf-lite)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite)
+    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS})
   else()
     target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LITE_LIBRARIES})
   endif()
 else()
   if(TARGET protobuf::libprotobuf)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf)
+    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf PRIVATE ${protobuf_ABSL_USED_TARGETS})
```

Comment:
`protobuf_ABSL_USED_TARGETS` is only needed when `("${Protobuf_VERSION}" VERSION_GREATER_EQUAL "4.22.0")`?

Reply:
This is a good question. I have test many cases:
1. If the protobuf is built in this project, the generated `.cmake` files will expose the dependency on ABSL libs, so it not necessary to use this `protobuf_ABSL_USED_TARGETS`, which means it's OK to be empty. 
2. `("${Protobuf_VERSION}" < "4.22.0")` there is no such dependency.
3. When protobuf is pre-installed and  when `("${Protobuf_VERSION}" >= "4.22.0")`, there could be no such generated `.cmake` files in pre-installed package and `find_package` only returns the protobuf libs only. In this case, we need to use this protobuf_ABSL_USED_TARGETS. Otherewise, there will be link errors.

Finnally, the conclusion is that `protobuf_ABSL_USED_TARGETS` is needed when protobuf is pre-installed and  when `("${Protobuf_VERSION}" VERSION_GREATER_EQUAL "4.22.0")`.